src/gui/qgsrelationeditorwidget.h

```diff
@@ -178,6 +180,20 @@ class GUI_EXPORT QgsRelationEditorWidget : public QgsCollapsibleGroupBox
 
     bool mShowLabel = true;
     bool mVisible = false;
+
+    /**
+     * Deletes the features
+     * \param featureids features to delete
+     * \since QGIS 3.00
+     */
+    void deleteFeatures( const QgsFeatureIds &featureids );
+
+    /**
+     * Unlinks the features
```

Comment:
"providing an empty list will use the current selection."

Reply:
I changed that. There are three methods now per function: `duplicateSelectedFeatures` (called by button), `duplicateFeature` (called by contextmenu), where `duplicateFeature` calls the `duplicateFeatures `with a list containing only one current feature and the `duplicateSelectedFeatures` with a list containing the selected features. It's not depending on the size of the `FeatureIds` if it's selected or not...